gluecodium/src/main/resources/templates/ffi/FfiCallbacksQueueImpl.mustache

```diff
@@ -51,19 +51,27 @@ CallbackQueue::enqueueIncoming(std::function<void()> func)
     return m_queue.back().done.get_future();
 }
 
-bool
-CallbackQueue::waitForIncoming()
+CallbackQueue::WaitResult
+CallbackQueue::waitForIncoming(const std::chrono::milliseconds& timeout)
 {
     std::unique_lock<std::mutex> lock(m_mutex);
 
-    m_notified.wait(lock, [this] { return m_has_incoming || m_is_closed; });
+    auto check_condition = [this] { return m_has_incoming || m_is_closed; };
+
+    if (timeout != std::chrono::milliseconds::zero()) {
```

Comment:
Inverting condition here would allow for if {} else if {} with slightly less nesting.